packages/@aws-cdk/cdk/lib/cross-environment-token.ts

```diff
@@ -0,0 +1,66 @@
+import { ArnComponents } from './arn';
+import { DeployTimeOrAssignedPhysicalName } from "./deploy-time-or-assigned-physical-name";
+import { PhysicalName } from './physical-name';
+import { IResource } from './resource';
+import { ResolveContext, Token } from './token';
+
+/**
+ * A Token that represents a reference that spans accounts and/or regions,
+ * and so requires the resources to have physical names.
+ * You should never need to interact with these directly,
+ * instead use the {@link ResourceIdentifiers} class.
+ * This class is private to the @aws-cdk/cdk package.
+ */
+export abstract class CrossEnvironmentToken extends Token {
+  private readonly resource: IResource;
+  private readonly physicalName: PhysicalName;
+
+  /**
+   * @param regularValue the value used when this is referenced NOT from a cross account and/or region Stack
+   * @param crossEnvironmentValue the value used when this is referenced from a cross account and/or region Stack
+   * @param resource the scope this reference is mastered in. Used to determine the owning Stack
+   * @param physicalName
+   * @param displayName a short name to be used in Token display
+   */
+  protected constructor(private readonly regularValue: string, private readonly crossEnvironmentValue: any,
```

Comment:
Change to keyword args (props)

Reply:
Can we keep it this way for now...? I don't really feel like defining 3 more interfaces for these. Remember that this class is private to the `@aws-cdk/cdk` package.